### PR TITLE
Table Inspector Improvements

### PR DESCRIPTION
**Pull request by ghislaineguerin**
### Modals Update

I've been improving modals across Mathesar to ensure consistency, better usability, and adherence to design guidelines. Here's a summary of the changes and improvements:

#### Alignment and Spacing Issues
- **Modal Headers:** The prominence of modal headers has been increased to ensure balance with the footers, as there was insufficient visual weight at the top.
- image
- The modal title size is now consistent and larger than content headings. This issue was especially noticeable in the Constraints modal, where the headers inside the modal were larger than the modal header.
image
- **Close Button Alignment:** The close button is now correctly aligned with the start of the title in all modals.
- image
- **Collapsing Headers:** In Constraint modals, collapsing headers were adjusted to maintain a consistent height, regardless of whether a button is present.
- image

#### Styling and Contrast Issues
- **Form Field Labels:** Improved the contrast of form field labels to distinguish them clearly from body text.
- image
- **Message Boxes:** Updated the style of message boxes to have a less boxy appearance, improving the flow of content and reducing vertical separations that make content harder to scan.
- image
- **Inline Forms and Descriptive Button Labels:** Inline forms were used to add foreign or unique keys, and redundant titles were removed. Instead, the save button labels were updated to indicate the action, such as adding a new object.
- image

#### Consistency Enhancements
- **Input Box Style:** Ensured consistency in using boxed input groups, applying the same style across different modals, such as import modals and database creation modals.
    - image
- **Modal Descriptions:** Standardized the placement and presentation of descriptions.xw

### Create Link Updates

The "Create Link" modal has been revised to enhance clarity and maintain consistency with other forms in the app.

- **Outcome Box Update**: Each outcome now has its own collapsible section containing a single, concise outcome statement. Previously, the outcome box listed all the input boxes for naming every element involved, which cluttered the interface. If the user agrees with the default outcomes, they won't need to interact with these input boxes. This change reduces visual distractions and makes it easier to review the outcomes which was the main goal for this special message box.
- **Updated Labels**: Labels have been revised to follow the new naming convention for new elements. For example, fields now use phrasing like "Name of new column" instead of "New column name".
- image
- image

### Color Scheme

- The color scheme has been updated to increase contrast. This change applies to the inspector panel, message boxes, and buttons.
image

### Record Summary Section

- The record summary section was updated for consistent language and to reduce redundant elements.
- Spacing between elements has been fixed, as some were too close together, such as the "Append Column" button and the input box.
- The tint for the record summary element, originally specified as dark yellow, lacked sufficient yellow and was adjusted from an orange tone to a more vibrant color.
- image

### Form Style Updates

- Labels have been made slightly bolder. Spacing between labels and input boxes has been updated for consistency with other parts of the app.
- image

### Button Sizes

- Button sizes have been updated so that all are using the "small" style when inside inspector.

### Spacing for empty states in tabs

- Spacing for empty states has been updated to be consistent across all tabs.
- image

### Links Styling
- Link card styling updated to match record link color shade. Direction icons moved outside circles to reduce visual weight.
- image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208365459259464

### CODE REVIEW

**Review by seancolsen — COMMENTED** [review]
This seems to be a visual regression: 

image

**Review by seancolsen — COMMENTED** [review]
I added some inline comments.

A couple other things:

- This seems wrong: 

    image


- Also I notice all your "run format" commits. I strongly suggest configuring Format on Save like this:

    image
    
    I find it to be a huge time saver!

    I leave it turned off at the "User" level and only enable it on a per-workspace basis.

**Review by kgodey — COMMENTED** [review]
(I only reviewed screenshots)

Great work on this, @ghislaineguerin! Looks so much better.

My only minor feedback is that the "In this table" and "From other tables" text in the Links section doesn't look like "label" text, it looks the same as the table name, etc. I'm not sure if this is a problem, I wouldn't block this PR on it, but wanted to flag in case you had any thoughts or ideas.

**Comment by ghislaineguerin** [comment]
> (I only reviewed screenshots)
> 
> Great work on this, @ghislaineguerin! Looks so much better.
> 
> My only minor feedback is that the "In this table" and "From other tables" text in the Links section doesn't look like "label" text, it looks the same as the table name, etc. I'm not sure if this is a problem, I wouldn't block this PR on it, but wanted to flag in case you had any thoughts or ideas.

I have updated the labels so that they use the same font-weight as other labels

image

File: mathesar_ui/src/components/form/Field.svelte [hunk 1]
```diff
@@ -23,20 +23,27 @@
...
 
 <FieldLayout>
   {#if label || $$slots.label}
-    <LabeledInput {label} {layout} {help}>
-      <slot name="label" slot="label" />
-      <slot name="help" slot="help" />
+    <LabeledInput {layout} {label}>
+      <svelte:fragment slot="label">
+        <slot name="label" />
+      </svelte:fragment>
```
[the remark this answers is not in the document]
Comment (ghislaineguerin) [reply mathesar_ui/src/components/form/Field.svelte]: I reverted this. I was trying to include another element inside the label slot but it was not a good idea.

**Comment by ghislaineguerin** [comment]
Closing this PR since we decided to consolidate all changes into one:. See conversation here: https://matrix.to/#/!UnujZDUxGuMrYdvgTU:matrix.mathesar.org/$yxAeaL_S4E1_HgmnpeZadGTOG9cTKzWOsFeRRkIE0XI?via=matrix.mathesar.org&via=matrix.org

**Pull request closed without merging**